packages/test/test-end-to-end-tests/src/test/compression.spec.ts

```diff
@@ -85,13 +96,65 @@ const compressionSuite = (getProvider) => {
 			if (provider.type === "TestObjectProviderWithVersionedLoad") {
 				this.skip();
 			}
+			await setupContainers();
 			const value = generateRandomStringOfSize(5);
 			localMap.set("testKey", value);
 
 			await provider.ensureSynchronized();
 			assert.strictEqual(localMap.get("testKey"), value);
 			assert.strictEqual(remoteMap.get("testKey"), value);
 		});
+
+		const messageGenerationOptions = generatePairwiseOptions<{
+			compressionAndChunking?: {
+				chunking: boolean;
+			};
+			grouping: boolean;
+		}>({
+			compressionAndChunking: [undefined, { chunking: false }, { chunking: true }],
+			grouping: [true, false],
+		});
+
+		messageGenerationOptions.forEach((option) => {
+			it(`Correctly processes messages: compression [${
+				option.compressionAndChunking !== undefined
+			}] chunking [${option.compressionAndChunking?.chunking === true}] grouping [${
+				option.grouping
+			}]`, async function () {
+				// TODO: Re-enable after cross version compat bugs are fixed - ADO:6287
+				if (provider.type === "TestObjectProviderWithVersionedLoad") {
+					this.skip();
+				}
+				await setupContainers({
+					compressionOptions:
+						option.compressionAndChunking === undefined
+							? undefined
+							: {
+									minimumBatchSizeInBytes: 10,
+									compressionAlgorithm: CompressionAlgorithms.lz4,
+							  },
+					chunkSizeInBytes:
+						option.compressionAndChunking?.chunking !== true ? undefined : 100,
+					enableGroupedBatching: option.grouping,
+				});
+				const values = [
+					generateRandomStringOfSize(100),
+					generateRandomStringOfSize(100),
+					generateRandomStringOfSize(100),
+				];
+				localDataObject.context.containerRuntime.orderSequentially(() => {
+					for (let i = 0; i < values.length; i++) {
+						localMap.set(`${i}`, values[i]);
+					}
+				});
+
+				await provider.ensureSynchronized();
+				for (let i = 0; i < values.length; i++) {
+					assert.equal(localMap.get(`${i}`), values[i]);
```

Comment:
Would this actually ever break?  Is it enough to just call `ensureSynchronized`? I guess it doesn't hurt...

Reply:
Just a simple e2e validation that combinations of compression, chunking, and op grouping don't break eventual consistency